File: src/main/java/wraith/fabricaeexnihilo/impl/BottleHarvestingImpl.java
package wraith.fabricaeexnihilo.impl;

import net.minecraft.block.BlockState;
import net.minecraft.block.Blocks;
import net.minecraft.item.ItemStack;
import wraith.fabricaeexnihilo.util.ItemUtils;

public final class BottleHarvestingImpl {

    public static ItemStack getResult(BlockState target) {
        return target.getBlock() != Blocks.SAND ? ItemStack.EMPTY : new ItemStack(ItemUtils.getExNihiloItem("salt_bottle"));
    }

}

File: src/main/kotlin/exnihilofabrico/impl/BottleHarvestingImpl.kt
package exnihilofabrico.impl

import exnihilofabrico.util.getExNihiloItem
import net.minecraft.block.BlockState
import net.minecraft.block.Blocks
import net.minecraft.item.ItemStack

object BottleHarvestingImpl {

    fun getResult(target: BlockState): ItemStack {
        return if(target.block != Blocks.SAND)
            ItemStack.EMPTY
        else
            ItemStack(getExNihiloItem("salt"))
    }

}